src/plugin.js

```diff
@@ -526,18 +536,20 @@ const contribAdsPlugin = function(options) {
           } else {
             this.state = 'ads-ready?';
           }
-          // When a new source is loaded into the player, we should remove the snapshot
-          // to avoid confusing player state with the new content's state
-          // i.e When new content is set, the player should fire the ended event
-          if (this.snapshot && this.snapshot.ended) {
-            this.snapshot = null;
-          }
```

Comment:
`contentupdate` should happen any time the source changes to a new source. With the new logic snapshot is nulled on contentupdate in any state, rather than just in `content-playback`, so that's the big change here.